src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -1,37 +1,101 @@
-  See the License for the specific language governing permissions and
-  limitations under the License.
---%>
-<!DOCTYPE html>
-<html>
-<head>
-  <title>Activity Feed</title>
-  <link rel="stylesheet" href="/css/main.css">
-</head>
-<body>
-
-  <nav>
-    <a id="navTitle" href="/">CodeU Chat App</a>
-    <a href="/conversations">Conversations</a>
-    <a href="/about.jsp">About</a>
-  </nav>
-
-  <div id="container">
-    <h1>Activity</h1>
-
-  		<p>This is the activity feed page</p>
-
-  </div>
-</body>
-</html>
+<%--
+  Copyright 2017 Google Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--%>
+<%@ page import="java.util.*"%>
+<%@ page import="java.util.Arrays"%>
+<%@ page import="java.time.*"%>
+<%@ page import="java.time.format.DateTimeFormatter"%>
+<%@ page import="codeu.model.data.Conversation"%>
+<%@ page import="codeu.model.data.User"%>
+<%@ page import="codeu.model.data.Message"%>
+<%@ page import="codeu.model.store.basic.UserStore"%>
+<%@ page import="java.time.format.FormatStyle"%>
+<%@ page import="codeu.model.store.basic.ConversationStore"%>
+<%@ page import="codeu.model.store.basic.MessageStore"%>
+<!DOCTYPE html>
+<html>
+<head>
+<title>Activity Feed</title>
+<link rel="stylesheet" href="/css/main.css">
+</head>
+<body>
+
+	<nav>
+		<a id="navTitle" href="/">CodeU Chat App</a> <a href="/conversations">Conversations</a>
+		<a href="/about.jsp">About</a>
+	</nav>
+
+	<div id="container">
+		<h1>Activity</h1>
+
+		<p>See what's happening!</p>
+		<ul>
+			<%
+				DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT).withLocale(Locale.US)
+						.withZone(ZoneId.systemDefault());
+				HashMap<Instant, HashMap<UUID, String>> instantByInstance = (HashMap<Instant, HashMap<UUID, String>>) request
+						.getAttribute("instantByInstance");
+				ArrayList<Instant> arrInstantsSorted = (ArrayList<Instant>) request.getAttribute("arrInstantsSorted");
+				if (instantByInstance == null) {
+					return;
+				} else {
+					for (Instant instant : arrInstantsSorted) {
+						String author;
+						Instant time;
+						String title;
+						for (Map.Entry<Instant, HashMap<UUID, String>> m : instantByInstance.entrySet()) {
+							if (m.getKey() == instant) {
+								HashMap<UUID, String> innerm = m.getValue();
+								for (Map.Entry<UUID, String> im : innerm.entrySet()) {
+									if (im.getValue() == "user") {
+										author = UserStore.getInstance().getUser(im.getKey()).getName();
+			%>

```

Comment:
I think it would make more sense to declare a userStore variable, messageStore variable and conversationStore variable instead of calling .getInstance() each time.